### DISCUSSION TITLE
Permissions slide doesn't move forward

### DISCUSSION THREAD

**Discussion by kamathprasad9** [AppIntro/AppIntro#1213] [Q&A] [answered]
I am trying to request some permissions from the user.

```
addSlide(
            AppIntroFragment.createInstance(
                "Requesting permissions,
                "This app requires a few permissions"
                imageDrawable = R.drawable.placeholder,
                backgroundColorRes = R.color.main_primary
            )
        )

askForPermissions(Constants.permissions, 3, true)
```

The `permissions` stored in Constants 
```
public final static String[] permissions = new String[]{Manifest.permission.WRITE_EXTERNAL_STORAGE,
            Manifest.permission.ACCESS_COARSE_LOCATION,
            Manifest.permission.ACCESS_FINE_LOCATION, Manifest.permission.INTERNET,
            Manifest.permission.RECORD_AUDIO, Manifest.permission.SYSTEM_ALERT_WINDOW,
            Manifest.permission.ACCESS_NETWORK_STATE, Manifest.permission.CAMERA};
```

The app asks for a couple of permissions (camera and location) and gets stuck on the slide as other permissions have not been explicitly granted. Am I doing anything wrong here? Do I need to modify my `permissions` array? Even if I change the `required` value to false, it still exhibits the same behavior. I believe I am following the [permissions documentation](https://github.com/AppIntro/AppIntro#permission)

**Comment by cortinico** [comment, marked as answer]
I'd suggest you check our example app, specifically here:
https://github.com/AppIntro/AppIntro/blob/7d329e7ecb2e52317b5194bf213000af6198deb6/example/src/main/java/com/github/appintro/example/ui/permissions/PermissionsIntro2.kt#L12-L45

It doesn't seems like you're doing anything wrong, but it's hard to tell without a reproducer.

**Reply by kamathprasad9** [reply]
Hi @cortinico thanks for the reply
I was able to get this fixed.

I had a question about the custom layouts that use TextViews. Is it necessary for them to use one of the text styles that belongs to AppIntro (eg. AppIntroDefaultText)? If I try to add a TextView, I get a error while inflating the textview, possibly due to theming issues.
